larq/quantized_variable.py

```diff
@@ -0,0 +1,308 @@
+        var = self.latent_variable.scatter_update(*args, **kwargs)
+        return _maybe_wrap(var, self.quantizer, self.precision)
+
+    def batch_scatter_update(self, *args, **kwargs):
+        var = self.latent_variable.batch_scatter_update(*args, **kwargs)
+        return _maybe_wrap(var, self.quantizer, self.precision)
+
+    def scatter_nd_sub(self, *args, **kwargs):
+        var = self.latent_variable.scatter_nd_sub(*args, **kwargs)
+        return _maybe_wrap(var, self.quantizer, self.precision)
+
+    def scatter_nd_add(self, *args, **kwargs):
+        var = self.latent_variable.scatter_nd_add(*args, **kwargs)
+        return _maybe_wrap(var, self.quantizer, self.precision)
+
+    def scatter_nd_update(self, *args, **kwargs):
+        var = self.latent_variable.scatter_nd_update(*args, **kwargs)
+        return _maybe_wrap(var, self.quantizer, self.precision)
+
+    def count_up_to(self, *args, **kwargs):
+        return self.latent_variable.count_up_to(*args, **kwargs)
+
+    def load(self, *args, **kwargs):
+        return self.latent_variable.load(*args, **kwargs)
+
+    @property
+    def dtype(self):
+        return self.latent_variable.dtype
+
+    @property
+    def name(self):
+        return self.latent_variable.name
+
+    @property
+    def _shared_name(self):
+        return self.latent_variable._shared_name
+
+    @property
+    def initializer(self):
+        return self.latent_variable.initializer
+
+    @property
+    def device(self):
+        return self.latent_variable.device
+
+    @property
+    def op(self):
+        return self.latent_variable.op
+
+    @property
+    def graph(self):
+        return self.latent_variable.graph
+
+    @property
+    def shape(self):
+        return self.latent_variable.shape
+
+    def get_shape(self):
+        return self.latent_variable.get_shape()
+
+    def _gather_saveables_for_checkpoint(self):
+        # By delegating this method to the wrapped variable, checkpoints with
+        # QuantizedVariables are identical to checkpoints with normal variables.
+        # Therefore models checkpointed with QuantizedVariables can be restored on
+        # models with normal variables, and vice versa.
+        return self.latent_variable._gather_saveables_for_checkpoint()
+
+    # TODO: Maybe encode the fact the variable is an QuantizedVariable in to_proto().
+    def to_proto(self, *args, **kwargs):
+        return self.latent_variable.to_proto(*args, **kwargs)
+
+    def from_proto(self, *args, **kwargs):
+        return self.latent_variable.from_proto(*args, **kwargs)
+
+    def _as_graph_element(self):
+        return self._quantize(self.latent_variable._as_graph_element())
+
+
+QuantizedVariable._OverloadAllOperators()
+tf.register_tensor_conversion_function(
+    QuantizedVariable, QuantizedVariable._dense_var_to_tensor
+)
+ops.register_dense_tensor_like_type(QuantizedVariable)
+
+
+def create_quantized_variable(variable, quantizer=None, precision=None):
```

Comment:
See above; I think this would make sense as a factory within the `QuantizedVariable` class.

Reply:
How do you feel about violating PEP8 and naming this function `QuantizedVariable` so it feels like a class when using it?

Reply:
It feels a bit wrong but I think it might make sense here.

Reply:
> It feels a bit wrong but I think it might make sense here.

Yeah I also feel this would be kinda wrong